Wrong CS2021 is reported for *.resources files

**Investigation**
A change in src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs made in https://github.com/dotnet/roslyn/commit/d23cc53f65501798ae46e8ab9f3aa74077a7586d causes a bug.

There, PathUtilities.IsValidFilePath is called for a fileName instead of a filePath. Inside it,
new FileInfo(fileName) is created, and there is no parent directory in fileName, so current directory is used. In our case current directory is a path to compiler binaries, and it is rather long. If path to compiler binaries length + file name length exceed 260 characters, then PathTooLongException is thrown and that results in CS2021 error.

Compiler errors should not depend on a path to compiler binaries, only on path to files being compiled.

**Version Used**: 
2.8 (from dev15.7)

**Steps to Reproduce**:
1. Place compiler binaries into a folder with a path length >=127 characters (say, C:\Users\SomeUserAtSomeCompany\AppData\Local\MyFolder1\~\Toolset\A367E15E0EE028AF\MyFolder1.Toolset.Roslyn.2.8.20180511.192024)
2. Make a resource file with a name length >= 135 characters (say, MyCompany.SomeNamespace.SubNamespace.MoreNamespace.OneMoreNamespace.EvenMoreNamespace.Control.ShowMeSomeOptionsOptionsPagePanel.resources) and place it in your project with a short folder name so that a full path length to resource file would not exceed 260 characters.
3. Compile your project with a said resource file using a compiler run from its folder from step 1.

**Expected Behavior**:
No errors are produced

**Actual Behavior**:
CSC : error CS2021: File name '...\MyCompany.SomeNamespace.SubNamespace.MoreNamespace.OneMoreNamespace.EvenMoreNamespace.Control.ShowMeSomeOptionsOptionsPagePanel.resources' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long

--- a/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs
+++ b/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs
@@ -1796,7 +1796,7 @@ namespace Microsoft.CodeAnalysis.CSharp
                 return null;
             }
 
-            if (fullPath == null || !PathUtilities.IsValidFilePath(fileName))
+            if (!PathUtilities.IsValidFilePath(fullPath))
             {
                 AddDiagnostic(diagnostics, ErrorCode.FTL_InvalidInputFileName, filePath);
                 return null;

--- a/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs
+++ b/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs
@@ -697,18 +697,16 @@ namespace Roslyn.Utilities
         /// 
         /// The more accurate way is to let the framework parse the path and throw on any errors.
         /// </summary>
-        /// <param name="path"></param>
-        /// <returns></returns>
-        public static bool IsValidFilePath(string path)
+        public static bool IsValidFilePath(string fullPath)
         {
             try
             {
-                if (string.IsNullOrEmpty(path))
+                if (string.IsNullOrEmpty(fullPath))
                 {
                     return false;
                 }
 
-                var fileInfo = new FileInfo(path);
+                var fileInfo = new FileInfo(fullPath);
                 return !string.IsNullOrEmpty(fileInfo.Name);
             }
             catch (Exception ex) when (

--- a/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs
+++ b/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs
@@ -706,7 +706,8 @@ namespace Roslyn.Utilities
                     return false;
                 }
 
-                Debug.Assert(IsAbsolute(fullPath));
+                // Uncomment when this is fixed: https://github.com/dotnet/roslyn/issues/19592
+                // Debug.Assert(IsAbsolute(fullPath));
 
                 var fileInfo = new FileInfo(fullPath);
                 return !string.IsNullOrEmpty(fileInfo.Name);

--- a/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs
+++ b/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs
@@ -706,6 +706,8 @@ namespace Roslyn.Utilities
                     return false;
                 }
 
+                Debug.Assert(IsAbsolute(fullPath));
+
                 var fileInfo = new FileInfo(fullPath);
                 return !string.IsNullOrEmpty(fileInfo.Name);
             }

--- a/src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb
+++ b/src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb
@@ -1625,7 +1625,7 @@ lVbRuntimePlus:
                 Return Nothing
             End If
 
-            If fullPath Is Nothing OrElse Not PathUtilities.IsValidFilePath(fileName) Then
+            If Not PathUtilities.IsValidFilePath(fullPath) Then
                 AddDiagnostic(diagnostics, ERRID.FTL_InvalidInputFileName, filePath)
                 Return Nothing
             End If

